aten/src/ATen/native/Normalization.cpp

```diff
@@ -19,6 +19,60 @@ static const int MIOPEN_DIM_MAX = 5;
 namespace at {
 namespace meta {
 
+static inline bool is_contiguous(const Tensor& t) {
+  return t.is_contiguous() || t.is_contiguous(at::MemoryFormat::ChannelsLast);
+}
+
+// For some ambiguous cases, it is possible a channels last contiguous Tensor has
+//   `suggest_memory_format` of Contiguous.
+// See https://github.com/pytorch/pytorch/issues/63224 for details.
+static inline MemoryFormat suggest_memory_format_contig(const Tensor& t) {
+  return t.is_contiguous() ? at::MemoryFormat::Contiguous : at::MemoryFormat::ChannelsLast;
+}
+
+TORCH_META_FUNC(native_batch_norm)(const at::Tensor & input,
+                                   at::OptionalTensorRef weight,
+                                   at::OptionalTensorRef bias,
+                                   at::OptionalTensorRef running_mean,
+                                   at::OptionalTensorRef running_var,
+                                   bool training, double momentum, double eps) {
+
+  const bool has_running_mean = (running_mean.has_value() && running_mean->defined());
+  const bool has_running_var = (running_var.has_value() && running_var->defined());
+  TORCH_CHECK(has_running_mean == has_running_var);
+
+  int64_t n_input = input.size(1);
+
+  bool all_contiguous = is_contiguous(input)
+      && (!weight->defined() || weight->is_contiguous())
+      && (!bias->defined() || bias->is_contiguous())
+      && running_mean->is_contiguous()
+      && running_var->is_contiguous();
+
+  // TODO(whc) using input.suggest_memory_format() passes test_ops.py,  fails test_nn.py (:9252, self.assertTrue(out.is_contiguous(memory_format=torch.channels_last)))
```

Comment:
Good catch!

It looks like that special handling isn't there in the existing cuda kernel though... https://github.com/pytorch/pytorch/blob/f9422e1c6b5dab15eacbbb0e1d4370b07db1e8e8/aten/src/ATen/native/cuda/Normalization.cu#L432. Maybe that means that the bug in the above issue was "fixed" only for cpu?

Reply:
@ngimel Could you weigh in? 
It looks like on cuda, we are just using the input memory format.  On CPU, we are checking _all_ the input tensors, and then making a decision.  Is this safe to land?  It seems like more work should be done to unify the cpu/cuda paths, but i'm hoping to not have to do it all in this PR.

Reply:
So this PR is changing cuda behavior st output will no longer propagate channels last if some of the other tensors are not contiguous? That's not great but we can live with this as long as we can fix it relatively soon (just by calling `.contiguous` on all those tensors in the cpu implementation).

Reply:
i think you're right, that would be a change in cuda behavior with this PR;  

Frankly, I'm not sure what the reasoning behind the current behavior on CPU side is.  Is it better to just make a simpler check (like the one cuda uses) and then change the unit tests?  or are we stuck mirroring whatever the current behavior is, to avoid it being a BC issue?